zhaquirks/tuya/ts0601_din_power.py

```diff
@@ -0,0 +1,171 @@
+"""Tuya Din Power Meter."""
+from zigpy.profiles import zha
+import zigpy.types as t
+from zigpy.zcl.clusters.general import Basic, Groups, Ota, Scenes, Time
+from zigpy.zcl.clusters.homeautomation import ElectricalMeasurement
+from zigpy.zcl.clusters.smartenergy import Metering
+
+from zhaquirks import Bus, LocalDataCluster
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.tuya import TuyaManufClusterAttributes, TuyaOnOff, TuyaSwitch
+
+TUYA_TOTAL_ENERGY_ATTR = 0x0211
+TUYA_CURRENT_ATTR = 0x0212
+TUYA_POWER_ATTR = 0x0213
+TUYA_VOLTAGE_ATTR = 0x0214
+TUYA_DIN_SWITCH_ATTR = 0x0101
+
+SWITCH_EVENT = "switch_event"
+
+
+class TuyaManufClusterDinPower(TuyaManufClusterAttributes):
+    """Manufacturer Specific Cluster of the Tuya Power Meter device."""
+
+    manufacturer_attributes = {
+        TUYA_TOTAL_ENERGY_ATTR: ("energy", t.uint16_t),
+        TUYA_CURRENT_ATTR: ("current", t.int16s),
+        TUYA_POWER_ATTR: ("power", t.uint16_t),
+        TUYA_VOLTAGE_ATTR: ("voltage", t.uint16_t),
+        TUYA_DIN_SWITCH_ATTR: ("switch", t.uint8_t),
+    }
+
+    def _update_attribute(self, attrid, value):
+        super()._update_attribute(attrid, value)
+        if attrid == TUYA_TOTAL_ENERGY_ATTR:
+            self.endpoint.device.energy_bus.listener_event(
```

Comment:
you can remove the `Bus` and you can get the electrical measurement cluster to do the updates like this: 

https://github.com/zigpy/zha-device-handlers/blob/10dbcc4d64fdf748989e6d72ea169bc7163eb6ad/zhaquirks/xiaomi/aqara/tvoc.py#L31

the ep_attribute to access is: `electrical_measurement`

https://github.com/zigpy/zigpy/blob/7c168c1c2ca2a338676e5956d2277af7c40971d6/zigpy/zcl/clusters/homeautomation.py#L82

I would remove the `Bus` and do this for all of the attrs.